discovery-provider/src/tasks/social_features.py

```diff
@@ -209,6 +212,12 @@ def add_track_repost(
         repost_user_id = event_args._userId
         repost_track_id = event_args._trackId
 
+        _, can_user_repost_track = premium_content_access_checker.check_access(
+            repost_user_id, repost_track_id, "track"
+        )
+        if not can_user_repost_track:
+            continue
```

Comment:
This would be a bit of an annoying change, but I think it make sense to have the access checker operate on a batch of items - otherwise, the # of DB roundtrips per block will scale with the # of likes/reposts in the block, which could bottleneck indexing

Reply:
implemented the change Piazza. great find
^ noting your comment Isaac. thanks